sklearn/multioutput.py

```diff
@@ -0,0 +1,212 @@
+        (n_cpus + 1 + n_jobs) are
+        used. Thus for n_jobs = -2, all CPUs but one are used.
+        
+        Note that parallel processing only occurs if there is
+        multiple classes within each target variable. 
+        It does each target variable in y in series.
+
+    Attributes
+    __________
+    estimator: Sklearn estimator: The base estimator used to constructe
+    the model. 
+        
+    """
+    
+    def __init__(self, estimator=None, n_jobs=1):
+
+        self.estimator = estimator
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y):
+        """ Fit the model to data. 
+        Creates a seperate model for each Response column. 
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+            Data.
+
+        y : (sparse) array-like, shape = [n_samples, p_targets]
+            Multi-class targets. An indicator matrix turns on multilabel
+            classification.
+
+        Returns
+        -------
+        self
+        """
+
+        # check to see that the data is numeric 
+
+        # check to see that the X and y have the same number of rows. 
+
+        # Calculate the number of classifiers
+        self._num_y = y.shape[1]
+
+        ## create a dictionary to hold the estimators. 
+        self.estimators_ ={}
+        
+        for i in range(self._num_y):
+            # init a new classifer for each and fit it. 
+            estimator = clone(self.estimator)   #make a fresh clone
+            ovr = OneVsRestClassifier(estimator,self.n_jobs)
+            self.estimators_[i] = ovr.fit(X,y[:, i])
+            
+        return self
+    
+    def predict(self, X):
+        """Predict multi-class multiple target variable using a model
+         trained for each target variable. 
+
+        Parameters
+        ----------
+        X : (sparse) array-like, shape = [n_samples, n_features]
+        Data.
+
+        Returns
+        -------
+        y : dict of [sparse array-like], shape = {predictors: n_samples}
+          or {predictors: [n_samples, n_classes], n_predictors}.
+            Predicted multi-class targets across multiple predictors.
+            Note:  entirely separate models are generated
+            for each predictor.
+        """
+        # check to see if the fit has been performed
+        check_is_fitted(self, 'estimators_')
+        
+        results = {}
+        for label, model_ in self.estimators_.iteritems():
+            results[label] = model_.predict( X)
+        return(results)
+    
+    def predict_proba(self, X):
+        """Probability estimates. This returns prediction probabilites 
+        for each class for each label in the form of 
+        a dictionary. 
+       
+        Parameters
+        ----------
+        X : array-like, shape = [n_samples, n_features]
+        Returns
+        -------
+        prob_dict (dict) A dictionary containing n_label sparse arrays 
```

Comment:
use standard formatting. I also have never seen `predict_proba` return this output type, but I've not been watching for a few months.